### PR TITLE
RabbitMQ SSL for irma-frontend (issue #112)

### PR DESCRIPTION
**Pull request by deloittem**
# RabbitMQ SSL for irma-frontend
## This is related to issue:
- Allow irma-frontend to use SSL with RabbitMQ: #112 

## Changelogs
- New folder SSL added in root repository, it will contain the certificates for irma-frontend
- Port configuration added in frontend.ini for easy switching between `NO SSL` / `SSL`
- SSL parameters added in frontend.ini : activation switch, SSL certificate names in "./ssl"

## Install
- Update **PORT** in `frontend.ini` from `5672` to `5671`. Except if you change the default port defined in the the RabbitMQ configuration template provided in "irma-brain/extras/rabbitmq"
- Update **activate_ssl** switch in `frontend.ini` to value `yes`
- Put the SSL certificates (CA_CERT, KEY_FILE, CERT_FILE) in "./ssl"
- Update **ca_certs, keyfile** and **certfile** in frontend.ini according to the filenames in "./ssl"

#### If you are switching to `ssl` from an already running `no_ssl` version, please do the following on irma-brain RabbitMQ server:
- sudo rabbitmqctl stop_app
- sudo rabbitmqctl reset
- sudo rabbitmqctl start_app
- create again the RabbitMQ vhosts, usernames and passwords:
- sudo ./extras/scripts/rabbitmq/rmq_adduser.sh probe probe mqprobe
- sudo ./extras/scripts/rabbitmq/rmq_adduser.sh brain brain mqbrain
- sudo ./extras/scripts/rabbitmq/rmq_adduser.sh frontend frontend mqfrontend

### CODE REVIEW

**Comment by guillaumededrie** [comment]
Hi @deloittem,
Thanks for the PR. I'm currently working on quarkslab/irma#20, so in the next days, will it be possible for you to re-open the pull request on the new repository (https://github.com/quarkslab/irma-web-ui)?

Regards,
Guillaume

**Comment by guillaumededrie** [comment]
Sorry, I shouldn't have closed this PR, it's not related to https://github.com/quarkslab/irma/issues/20.

We will take a look at it shortly, thanks!

File: config/parser.py [hunk 1]
```diff
@@ -87,6 +88,12 @@
         ('clean_db_cron_minute', TemplatedConfiguration.string, '0'),
         ('clean_db_cron_day_of_week', TemplatedConfiguration.string, '*'),
     ],
+    'ssl_config': [
+        ('activate_ssl', TemplatedConfiguration.string, "no"),
```
Comment (ch0k0bn) [inline config/parser.py]: maybe use a TemplatedConfiguration.boolean

**Comment by ch0k0bn** [comment]
merged

**Pull request closed without merging**